### PR TITLE
Max mediation

### PR DESCRIPTION
**Pull request by Nodrex**

### CODE REVIEW

File: app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    ActivityMaxRecyclerViewBinding binding;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+	super.onCreate(savedInstanceState);
+
+	binding = ActivityMaxRecyclerViewBinding.inflate(getLayoutInflater());
+	setContentView(binding.getRoot());
+
+	//Initialize AppLovinSdk
```
Comment (iozdova) [inline app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java:41]: The comments suppose to be like a tutorial with steps.

File: app/src/main/java/com/appharbr/example/app/max/MaxBannerActivity.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+	//      **** (4) ****
+	//Star loading Ads
+	binding.maxAdViewBanner.loadAd();
+
+	//Optionally set auto refresh of ads
+	binding.maxAdViewBanner.startAutoRefresh();
+    }
+
+    private final AHListener ahListener = (view, unitId, adFormat, reasons)
```
Comment (iozdova) [inline app/src/main/java/com/appharbr/example/app/max/MaxBannerActivity.java]: Our listener should be at the end of the class.

File: app/src/main/java/com/appharbr/example/app/max/MaxInterstitialActivity.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	@Override
+	public void onAdDisplayFailed(MaxAd ad, MaxError error) {
+	    Log.d("LOG", "Max - onAdDisplayFailed");
+	}
+    };
+
+    AHListener ahListener = (view, unitId, adFormat, reasons)
+	    -> Log.d("LOG", "GeoEdge - onAdBlocked for: " + unitId + ", reason: " + Arrays.toString(reasons));
```
Comment (iozdova) [inline app/src/main/java/com/appharbr/example/app/max/MaxInterstitialActivity.java]: Please change GeoEdge to AppHarbr

File: app/src/main/java/com/appharbr/example/app/max/MaxNativeAdActivity.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	    public void onNativeAdLoaded(@Nullable MaxNativeAdView maxNativeAdView, MaxAd maxAd) {
+		Log.d("LOG", "Max - onNativeAdLoaded");
+
+		//hiding progressBar cause ad is already loaded
+		binding.progressBar.setVisibility(View.GONE);
+
+		//Check loaded Max native ad from AppHarbr if it needs to be blocked
+		AdResult adResult = AppHarbr.shouldBlockNativeAd(AdSdk.MAX, maxAd);
+		if (adResult.getAdStateResult() != AdStateResult.BLOCKED) {
+		    Log.d("LOG", "**************************** AppHarbr Permit to Display Max Interstitial ****************************");
```
Comment (iozdova) [inline app/src/main/java/com/appharbr/example/app/max/MaxNativeAdActivity.java]: It's not Interstitial. Change the log to Native

File: app/src/main/java/com/appharbr/example/app/max/MaxNativeRecyclerViewActivity.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+	    public void onAdClicked(MaxAd maxAd) {}
+
+	    public void onAdRevenuePaid(MaxAd maxAd) {}
+	});
+
+	ahMaxRecyclerAdapter.getAdPlacer().setAdSize(-1, AppLovinSdkUtils.dpToPx(this, 200));
+
+	binding.adRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+	binding.adRecyclerView.setAdapter(ahMaxRecyclerAdapter);
```
Comment (iozdova) [inline app/src/main/java/com/appharbr/example/app/max/MaxNativeRecyclerViewActivity.java:70]: Add a comment here that the publisher using our wrapper adapter.

File: app/src/main/java/com/appharbr/example/app/max/MaxRewardedActivity.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	public void onAdLoaded(MaxAd ad) {
+	    Log.d("LOG", "Max - onAdLoaded");
+	    binding.progressBar.setVisibility(View.GONE);
+	    if (maxRewardedAd.isReady()) {
+		checkAd();
+	    }
+	}
+
+	private void checkAd() {
+	    final AdStateResult interstitialState = AppHarbr.getInterstitialState(maxRewardedAd);
```
Comment (iozdova) [inline app/src/main/java/com/appharbr/example/app/max/MaxRewardedActivity.java]: Should use the public method AppHarbr.getRewardedState instant of getInterstitialState.

**Review by iozdova — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java (hunk 1 above)
Comment (Nodrex) [reply app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java:41]:
```
//      **** (1) **** 
//      ApplovinSdk initialization
```

Like this?

Comment (iozdova) [reply app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java:41]: I think we do not need to add the Initialization as the first step.
They should do it anyway regardless of the ad format.

**Review by iozdova — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java (hunk 1 above)
Comment (Nodrex) [reply app/src/main/java/com/appharbr/example/app/max/MaxBannerRecyclerViewActivity.java:41]: aa you mean we do not need to mark comments with stars and numbers? 
So to say we do not need to include this as step?

**Review by iozdova — APPROVED** [review]

**Pull request merged**